aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/APIGatewayProxyRequestEvent.java

```diff
@@ -951,7 +953,15 @@ public APIGatewayProxyRequestEvent withHttpMethod(String httpMethod) {
      * @param headers The headers sent with the request
      */
     public void setHeaders(Map<String, String> headers) {
-        this.headers = headers;
+        if (headers == null || headers.isEmpty()) {
+            this.headers = null;
+            return;
+        }
+
+        if (this.headers == null) {
+            this.headers = new HttpHeaders<>();
+        }
+        this.headers.putAll(headers);
```

Comment:
This is a breaking change for consumers using `setHeaders` and expecting it to override all previous values